listenbrainz/webserver/views/api_compat.py

```diff
@@ -272,47 +272,61 @@ def record_listens(request, data):
     # With corrections than the original submitted listen.
     doc, tag, text = Doc().tagtext()
     with tag('lfm', status='ok'):
-        with tag('nowplaying' if listen_type == 'playing_now' else 'scrobbles'):
-
-            for origL, augL in zip(list(lookup.values()), augmented_listens):
-                corr = defaultdict(lambda: '0')
-
-                track = augL['track_metadata']['track_name']
-                if origL['track'] != augL['track_metadata']['track_name']:
-                    corr['track'] = '1'
-
-                artist = augL['track_metadata']['artist_name']
-                if origL['artist'] != augL['track_metadata']['artist_name']:
-                    corr['artist'] = '1'
-
-                ts = augL['listened_at']
-
-                albumArtist = artist
-                if origL.get('albumArtist', origL['artist']) != artist:
-                    corr['albumArtist'] = '1'
-
-                album = augL['track_metadata'].get('release_name', '')
-                if origL.get('album', '') != album:
-                    corr['album'] = '1'
-
-                with tag('scrobble'):
-                    with tag('track', corrected=corr['track']):
-                        text(track)
-                    with tag('artist', corrected=corr['artist']):
-                        text(artist)
-                    with tag('album', corrected=corr['album']):
-                        text(album)
-                    with tag('albumArtist', corrected=corr['albumArtist']):
-                        text(albumArtist)
-                    with tag('timestamp'):
-                        text(ts)
-                    with tag('ignoredMessage', code="0"):
-                        text('')
+        if listen_type == 'playing_now':
+            with tag('nowplaying'):
+                text(create_response_for_recorded_listens(lookup, augmented_listens))
+        else:
+            # At present listens are always accepted by LB.
+            # So ignored will be always '0' which is not the case with Last.FM.
+            accepted = str(len(lookup))
+            with tag('scrobbles', accepted=accepted, ignored='0'):
+                text(create_response_for_recorded_listens(lookup, augmented_listens))
 
     return format_response('<?xml version="1.0" encoding="utf-8"?>\n' + yattag.indent(doc.getvalue()),
                            output_format)
 
 
+def create_response_for_recorded_listens(lookup, augmented_listens):
```

Comment:
A docstring explaining what this function does would be nice to have. The names `lookup` and `augmented_listens` also don't explain much.